SCF-Assignment-7/SparseMatrix.java

```diff
@@ -0,0 +1,222 @@
+package SparseMatrix;
+
+/**
+ * 
+ * @author Arjita
+ *
+ */
+public class SparseMatrix {
+	private int[][] sparseMatrix;// 2D array to store non-zero elements of
+									// sparse matrix
+	private int numberOfRows;// number of rows of input array
+	private int numberOfCol;// number of columns of input array
+
+	/**
+	 * SparseMatrix is constructor of class which will initialise the sparse
+	 * matrix array in which there will be two columns, first column will store
+	 * value and second column will store the index This will decrease the space
+	 * complexity.
+	 * 
+	 * @param inputArray
+	 *            input sparse matrix
+	 */
+	public SparseMatrix(int[][] inputArray) {
+		numberOfRows = inputArray.length;// number of rows of array
+		numberOfCol = inputArray[0].length;// number of column of array
+		int countElements = 0;// count the non zero elements of input matrix
+		for (int i = 0; i < inputArray.length; i++) {
+			for (int j = 0; j < inputArray[i].length; j++) {
+				if (inputArray[i][j] != 0) {
+					countElements++;
+				}
+			}
+		}
+		sparseMatrix = new int[countElements][2];// first column will store
+													// value, second will store
+													// the index of that value
+		int count = 0;
+		for (int i = 0; i < numberOfRows; i++) {
+			for (int j = 0; j < numberOfCol; j++) {
+				if (inputArray[i][j] != 0) {
+					sparseMatrix[count][0] = inputArray[i][j];
+					sparseMatrix[count][1] = (i * numberOfCol) + j;// index will
+																	// be stored
+																	// with
+																	// respective
+																	// values
+					count++;
+				}
+			}
+		}
+		// display(sparseMatrix);
+	}
+
+	public int[][] getSparseMatrix() {
+		return sparseMatrix;
+	}
+
+	/**
+	 * transposeMatrix will transpose the sparse matrix the complexity of
+	 * transpose function will be O(n) where n is no. of non zero elements
+	 * 
+	 * @return transpose of sparse matrix
+	 */
+	public int[][] transposeMatrix() {
+		int index, newIndex;// variables for calculation of index in sparse and
+							// transpose matrix
+		int row, column;// variables for calculation of location of respective
+						// value in sparse matrix through index
+		int[][] sparseArray = this.getSparseMatrix();
+		int[][] transposeArray = new int[sparseArray.length][2];
+		for (int i = 0; i < sparseArray.length; i++) {
+			transposeArray[i][0] = sparseArray[i][0];// store the value as it is
+			index = sparseArray[i][1];// fetch the index of i th value
+			row = index / numberOfCol;
+			column = index % numberOfCol;
+			newIndex = column * numberOfRows + row;// new index for transpose
+													// array
+			transposeArray[i][1] = newIndex;
+		}
+		display(transposeArray);
+		int[][] arr = sparseToNormal(transposeArray, numberOfCol, numberOfRows);
+		return arr;

```

Comment:
We can combine the above 2 lines like:
return sparseToNormal(transposeArray, numberOfCol, numberOfRows);